File: src/test/java/net/haesleinhuepf/clij/ops/examples/Rotate.java

package net.haesleinhuepf.clij.ops.examples;

import java.io.IOException;

import org.junit.Test;

import net.haesleinhuepf.clij.ops.CLIJ_close.CLIJ_close;
import net.haesleinhuepf.clij.ops.CLIJ_pull.CLIJ_pull;
import net.haesleinhuepf.clij.ops.CLIJ_push.CLIJ_push;
import net.haesleinhuepf.clij.ops.CLIJ_rotateLeft.CLIJ_rotateLeft;
import net.imagej.ImageJ;

public class Rotate {

	@Test
	public void run() throws IOException {
		ImageJ ij = new ImageJ();
		Object input = ij.io().open(getClass().getResource("/samples/t1-head.tif")
			.getPath());
		ij.ui().show("input", input);
		// push image to GPU
		Object inputGPU = ij.op().run(CLIJ_push.class, input);
		// flip
		Object rotated = ij.op().run(CLIJ_rotateLeft.class, inputGPU);
		// show result
		Object result = ij.op().run(CLIJ_pull.class, rotated);
		ij.ui().show("rotated", result);

		// cleanup
		ij.op().run(CLIJ_close.class, inputGPU);
		ij.op().run(CLIJ_close.class, rotated);
		ij.op().run(CLIJ_close.class);

	}

	public static void main(String... args) throws IOException {
		Rotate task = new Rotate();
		task.run();
	}
}

File: src/test/resources/jython/rotate.py
# CLIJ example ImageJ Ops jython: rotate.py
#
# This macro shows how stacks can be rotated in the GPU.
#
# Author: Robert Haase, rhaase@mpi-cbg.de
# Author: Deborah Schmidt, frauzufall@mpi-cbg.de
# May 2019
# ---------------------------------------------

#@ IOService io
#@ UIService ui
#@ OpService ops

import os
import inspect
def getResource(file):
    return os.path.dirname(os.path.abspath(inspect.getsourcefile(lambda:0))) + "/" + file


input = io.open(getResource("../samples/t1-head.tif"))
ui.show("input", input)
# push image to GPU
inputGPU = ops.run("CLIJ_push", input)

rotated = ops.run("CLIJ_rotateLeft", inputGPU)
# show result
result = ops.run("CLIJ_pull", rotated)
ui.show("rotated", result)

#cleanup
ops.run("CLIJ_close", inputGPU)
ops.run("CLIJ_close", rotated)
ops.run("CLIJ_close")
